variables.tf

```diff
@@ -259,3 +253,31 @@ variable "tags" {
   default     = null
   description = "(Optional) Tags of the resource."
 }
+
+variable "vnet" {
+  type = object({
+    resource_id = optional(string, null)
+    subnets = map(object({
+      name              = string
+      address_prefixes  = list(string)
+      service_endpoints = optional(list(string), [])
+      nsg_id            = optional(string, null)
+    }))
+    address_space       = list(string)
+    resource_group_name = optional(string, null)
+  })
+  default = {
+    subnets = {
+      "aisubnet" = {
+        name             = "aisubnet"
+        address_prefixes = ["10.0.1.0/24"]
```

Comment:
How do we ensure these default addresses will not conflict with other subnets? Perhaps defaults only make sense for completely new vnets

Reply:
My understanding is that the default is only used when there are no subnets provided, so there shouldn't be any conflicts with these.